Allow hiding new/deleted formula/cask sections on update

eugenesvk opened the pull request:
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally? (see errors below)

-----

This information is mostly noise, so added an env var to allow to silence it

closes https://github.com/orgs/Homebrew/discussions/4590

tyrann0us commented on docs/Manpage.md:
```diff
@@ -747,6 +747,8 @@ Fetch the newest version of Homebrew and all formulae from GitHub using `git`(1)
 * `-f`, `--force`:
   Always do a slower, full update check (even if unnecessary).
 
+If `HOMEBREW_NO_UPDATE_REPORT_FORMULAE`/`HOMEBREW_NO_UPDATE_REPORT_CASKS` is set, `brew update` will not show the list of newly added formulae/casks respectively
```
```suggestion
If `HOMEBREW_NO_UPDATE_REPORT_FORMULAE`/`HOMEBREW_NO_UPDATE_REPORT_CASKS` is set, `brew update` will not show the list of newly added formulae/casks, respectively.
```

tyrann0us reviewed (commented):
Thanks for this! Does it also make sense to make disabling the _deleted_ formulae/casks lists possible?

eugenesvk commented:
as far as I undersand, deleted section reports only those formulas that the user has installed, so it's more relevant information (unless `HOMEBREW_UPDATE_REPORT_ALL_FORMULAE` is set)

But maybe it still makes sense to silence those as well, so also added that option

MikeMcQuaid commented on Library/Homebrew/cmd/update-report.rb:
```diff
@@ -749,12 +749,12 @@ def dump(auto_update: false)
 
-    dump_new_formula_report
-    dump_new_cask_report
+    dump_new_formula_report if ENV["HOMEBREW_NO_UPDATE_REPORT_FORMULAE"].blank?
```
Define this in `env_config.rb` instead and use the generated `Homebrew.no_update_report*?` methods

MikeMcQuaid commented on docs/Manpage.md:
```diff
@@ -747,6 +747,9 @@ Fetch the newest version of Homebrew and all formulae from GitHub using `git`(1)
 * `-f`, `--force`:
   Always do a slower, full update check (even if unnecessary).
 
+If `HOMEBREW_NO_UPDATE_REPORT_FORMULAE`/`HOMEBREW_NO_UPDATE_REPORT_CASKS` is set, `brew update` will not show the list of newly added formulae/casks, respectively.
```
Define this in `env_config.rb` instead.

eugenesvk replied:
`boolean:     true` there means variable type, not default value, correct?

eugenesvk replied:
moved it to env_config

MikeMcQuaid reviewed (commented):
@eugenesvk Thanks for the PR! 

Would a single variable that hid both meet both your needs?

@eugenesvk @tyrann0us

MikeMcQuaid commented:
> Does it also make sense to make disabling the _deleted_ formulae/casks lists possible?

@tyrann0us Can you explain why you want to hide this? As @eugenesvk said, it's only those that you have installed so seems like pretty important information.

eugenesvk commented:
A single var would be fine for me, just created 2 in case someone has more refined silencing tastes :)

tyrann0us commented:
Hi @MikeMcQuaid,

> Would a single variable that hid both meet both your needs?

Yes, that would work fine for me. I'd actually prefer that.

>  Can you explain why you want to hide this? As @eugenesvk said, it's only those that you have installed so seems like pretty important information.

TBH, I wasn't aware that this section only lists deleted formulae/casks that the user has installed. 🙈
Totally agree that this is important information, so please disregard that comment. 

Thanks!

MikeMcQuaid commented:
@eugenesvk Thanks. Do you personally want deleted formulae hidden or just new ones?

> TBH, I wasn't aware that this section only lists deleted formulae/casks that the user has installed. 🙈

Thanks. Could clarify the title of that section perhaps 👍🏻

MikeMcQuaid commented on Library/Homebrew/cmd/update-report.rb:
```diff
@@ -749,12 +749,12 @@ def dump(auto_update: false)
     dump_renamed_cask_report if report_all
-    dump_deleted_formula_report(report_all)
-    dump_deleted_cask_report(report_all)
+    dump_deleted_formula_report(report_all) if ENV["HOMEBREW_NO_UPDATE_REPORT_FORMULAE_DELETED"].blank?
```
Given discussion: can remove this bit.

eugenesvk replied:
removed

eugenesvk commented:
> Do you personally want deleted formulae hidden or just new ones?
> 
> >

 Do you personally want deleted formulae hidden or just new ones?

No, to me it's important to know of some formula I have disappears. Personally I'm fine having a single var that only suppresses updates, and it seems you and @tyrann0us too, so will change to 1

MikeMcQuaid commented on Library/Homebrew/env_config.rb:
```diff
@@ -327,6 +327,22 @@ module EnvConfig
                      "from running `brew install` <formula> or `brew upgrade` <formula>.",
         boolean:     true,
       },
+      HOMEBREW_NO_UPDATE_REPORT_FORMULAE:    {
+        description: "If set, `brew update` will not show the list of newly added formulae.",
+        boolean:     true,
+      },
+      HOMEBREW_NO_UPDATE_REPORT_CASKS:    {
+        description: "If set, `brew update` will not show the list of newly added casks.",
+        boolean:     true,
+      },
+      HOMEBREW_NO_UPDATE_REPORT_FORMULAE_DELETED:    {
+        description: "If set, `brew update` will not show the list of deleted formulae.",
+        boolean:     true,
+      },
+      HOMEBREW_NO_UPDATE_REPORT_CASKS_DELETED:    {
+        description: "If set, `brew update` will not show the list of deleted casks.",
+        boolean:     true,
+      },
```
```suggestion
      HOMEBREW_NO_UPDATE_REPORT_NEW:    {
        description: "If set, `brew update` will not show the list of newly added formulae or casks.",
        boolean:     true,
      },
```

(may need to run `brew style --fix` and split the string if it's too long)

MikeMcQuaid commented on Library/Homebrew/cmd/update-report.rb:
```diff
@@ -749,12 +749,12 @@ def dump(auto_update: false)
 
-    dump_new_formula_report
-    dump_new_cask_report
+    dump_new_formula_report if ENV["HOMEBREW_NO_UPDATE_REPORT_FORMULAE"].blank?
+    dump_new_cask_report if ENV["HOMEBREW_NO_UPDATE_REPORT_CASKS"].blank?
```
```suggestion
    if Homebrew::EnvConfig.no_update_report_new?
      dump_new_formula_report
      dump_new_cask_report
    end
```

MikeMcQuaid replied:
@eugenesvk check out this suggestion

eugenesvk replied:
yes, the typed env is much nicer, thanks

MikeMcQuaid commented on Library/Homebrew/cmd/update-report.rb:
```diff
@@ -749,12 +749,12 @@ def dump(auto_update: false)
     dump_renamed_cask_report if report_all
-    dump_deleted_formula_report(report_all)
-    dump_deleted_cask_report(report_all)
+    dump_deleted_formula_report(report_all) if ENV["HOMEBREW_NO_UPDATE_REPORT_FORMULAE_DELETED"].blank?
+    dump_deleted_cask_report(report_all) if ENV["HOMEBREW_NO_UPDATE_REPORT_CASKS_DELETED"].blank?
```
```suggestion
    dump_deleted_formula_report(report_all)
    dump_deleted_cask_report(report_all)
```

eugenesvk replied:
sry missed this change, already added

MikeMcQuaid reviewed (approved):
Thanks so much for your repeated contribution! Without people like you submitting PRs we couldn't run this project. You rock, @eugenesvk!
